### ISSUE TITLE
Character encoding of non-English characters lost

### ISSUE THREAD

**Issue by christophpre** [Asana/java-asana#84]
Hi,

I'm currently building an app with your library and have an issue with the encoding of non-English characters.

In Asana I created a task with the following name: **test task ü ä ö ß**

When I request the task with java-asana and the following code:

```Java
Client client = Client.oauth(app);
client.tasks.findById(taskId).execute();
```

the encoding of the characters is wrong: 
![image](https://user-images.githubusercontent.com/6893212/56348813-8b99a800-61c7-11e9-8e4a-30cb9f3ea228.png)

This problem appears not only for tasks, but for all Strings retrieved from Asana via the java-asana library. If I query over your REST endpoints (.i.e. https://app.asana.com/api/1.0/users), the encoding is fine. So it seems like an issue within the Java library.

**java-asana Version:** 0.8.1

**Comment by rossgrambo** [comment]
Hey @christophpre! 

Thank you for bringing this to our attention. I've added a task to do some discovery on this to see if I can find a solution. 

We're doing a deploy to this library soon, and I'll do a little bit of debugging on this to see if I can include a fix in to this deploy, but if I can't find a solution during this sprint, this will have to wait until our next deploy.

**Comment by christophpre** [comment]
@rossgrambo I think I have narrowed this problem down to the following:

In  the class **ItemRequest.java** there needs to be additional UTF-8 encoding in order to ensure prober conversion, when parsing the JSON Response:

**Before**

```Java
public ResultBody<T> executeRaw() throws IOException {
        HttpResponse response = this.client.request(this);
        return Json.getInstance().fromJson(
                new InputStreamReader(response.getContent()),
                new TypeToken<ResultBody<T>>() {
                }.where(
                        new TypeParameter<T>() {
                        },
                        this.elementClass
                ).getType()
        );
    }
```
**After**
```Java
public ResultBody<T> executeRaw() throws IOException {
        HttpResponse response = this.client.request(this);
        return Json.getInstance().fromJson(
                new BufferedReader(new InputStreamReader(response.getContent(), StandardCharsets.UTF_8)),
                new TypeToken<ResultBody<T>>() {
                }.where(
                        new TypeParameter<T>() {
                        },
                        this.elementClass
                ).getType()
        );
    }
```

Try with the following test method in **ClientTest.java**:
```Java
@Test
public void testClientGetWithNonEnglishCharacters() throws IOException
{
  dispatcher.registerResponse("GET", "http://app/users/me").code(200).content("{ \"data\": { \"name\": \"öäüßs\" }}");
  assertEquals("öäüßs", client.users.me().execute().name);
}
```

I'll include a pull-request for these changes.

**Comment by rossgrambo** [comment]
Oh you're awesome!

Thank you. I'll take a look at the PR and try to include it in the next version.

**Comment by christophpre** [comment]
No Problem. I just created the pull-request with changes for the **ItemRequest.java** and **CollectionRequest.java**. I could not find another usage of InputStreamReader in the project, so I'm not sure if there are any other parts with this problem. 

I also saw that in some test methods the order of the assertion parameters is wrong. I fixed this in the ClientTest.java class. There may be wrong usages in other test classes (e.g. **ResourcesTest.java**), but I have not looked through all of them. For instance:

```Java
  //before
  assertEquals(client.users.me().execute().name, "me");
  //after
  assertEquals("me", client.users.me().execute().name);
```
